kion/internal/kionclient/helper.go

```diff
@@ -251,17 +254,31 @@ func FieldsChanged(iOld interface{}, iNew interface{}, fields []string) (map[str
 }
 
 func OptionalBool(d *schema.ResourceData, fieldname string) *bool {
-	if v, ok := d.Get(fieldname).(bool); ok {
-		return &v
+	b, ok := d.GetOkExists(fieldname)
+	if !ok {
+		return nil
 	}
-	return nil
+
+	ret, ok := b.(bool)
+	if !ok {
+		return nil
+	}
+
+	return &ret
 }
 
 func OptionalInt(d *schema.ResourceData, fieldname string) *int {
-	if v, ok := d.Get(fieldname).(int); ok {
-		return &v
+	v, ok := d.GetOkExists(fieldname)
+	if !ok {
+		return nil
 	}
-	return nil
+
+	ret, ok := v.(int)
+	if !ok {
+		return nil
+	}
+
+	return &ret
 }
```

Comment:
Can you confirm these changes are just undoing the changes made in https://github.com/kionsoftware/terraform-provider-kion/pull/48/files#diff-bfa01f2b893b57e61ed2a4830530ac2bbdb67e8283c6adb2ec0ea41c8210afccL253 ?

I know we discussed these in the past.  Did you find problems with the changes?

Reply:
I did not find no problems, but I remember your reasoning about why it existed, which made sense.  I marked it to be ignored in the linting as well.